test/api_compatibility.js

```diff
@@ -121,13 +89,8 @@ module.exports = function apiCompatibilityChecks(createTracer) {
             });
```

Comment:
we could do it in different PR or in this one, but it seems none of the functions are actually being tested. For example, I would expect tests like:

  * each method should be called at least once by the harness
  * tracer() returns something that has the Tracer methods on it
  * methods like setTag() return something on which setTag can be called again (chaining)
  * get-baggage returns can return what was set with set-baggage (in Python we made this controlled by a parameter to the harness because noop implementation does not store the baggage)

Before we use this harness with our tracer, we need these tests implemented.